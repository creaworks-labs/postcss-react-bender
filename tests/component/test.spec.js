var { processFile } = require("../utils");

describe('component styles', () => {
  it("process basic component styles", async () => {
    const fixture = require.resolve('./fixtures/basic.scss')
  
    const result = await processFile(fixture);
  
    expect(result.bender).toMatchSnapshot();
    expect(result.warnings().length).toBe(0);
  });
  
  it("process nested component based styles", async () => {
    const fixture = require.resolve('./fixtures/nested.scss')
  
    const result = await processFile(fixture);
  
    expect(result.bender).toMatchSnapshot();
    expect(result.warnings().length).toBe(0);
  });
  
  it("process asterisk-ed component alternate styles", async () => {
    const fixture = require.resolve('./fixtures/asterisk.scss')
  
    const result = await processFile(fixture);
  
    expect(result.bender).toMatchSnapshot();
    expect(result.warnings().length).toBe(0);
  });
  
})
