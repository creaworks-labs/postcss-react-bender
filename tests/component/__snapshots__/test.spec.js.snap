// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component styles process asterisk-ed component alternate styles 1`] = `
Object {
  "Text": Object {
    "*": Object {
      "paddingVertical": 20,
    },
    "*.container": Object {
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
    },
    "backgroundColor": "gray",
  },
}
`;

exports[`component styles process basic component styles 1`] = `
Object {
  "View": Object {
    "backgroundColor": "red",
    "height": 50,
    "width": 100,
  },
}
`;

exports[`component styles process nested component based styles 1`] = `
Object {
  "Screen": Object {
    ".square": Object {
      "alignItems": "center",
      "backgroundColor": "yellow",
      "height": 55,
      "justifyContent": "center",
      "marginTop": "10pt",
      "width": 25,
    },
    "Text.underline": Object {
      "fontSize": 20,
      "textDecorationColor": "black",
      "textDecorationLine": "underline",
      "textDecorationStyle": "solid",
    },
    "View": Object {
      "backgroundColor": "red",
      "height": 50,
      "width": 100,
    },
    "fontSize": 25,
  },
  "Text": Object {
    ".underline": Object {
      "textDecorationColor": "black",
      "textDecorationLine": "underline",
      "textDecorationStyle": "solid",
    },
  },
}
`;
